mypy/semanal.py

```diff
@@ -6090,7 +6104,12 @@ def visit_set_comprehension(self, expr: SetComprehension) -> None:
     def visit_dictionary_comprehension(self, expr: DictionaryComprehension) -> None:
```

Comment:
And here as well? Probably worth extracting into a helper method

Reply:
extracted it to a small method and apply the check on both dict and list comprehensions visits